posthog/api/team.py

```diff
@@ -64,7 +69,28 @@ class Meta:
             "created_at",
             "updated_at",
             "ingested_event",
+            "effective_membership_level",
+        )
+
+    def get_effective_membership_level(self, team: Team) -> Optional[int]:
+        requesting_parent_membership: OrganizationMembership = OrganizationMembership.objects.only("id", "level").get(
+            organization_id=team.organization_id, user=self.context["request"].user
         )
+        if not settings.EE_AVAILABLE:
+            # Per-project access not available
+            return requesting_parent_membership.level
+        from ee.api.explicit_team_member import ExplicitTeamMembership
+
+        try:
+            team_membership = ExplicitTeamMembership.objects.only("level").get(
+                team=team, parent_membership=requesting_parent_membership
+            )
+            return max(requesting_parent_membership.level, team_membership.level)
```

Comment:
Repeated logic from a model, we should have this in a single place to avoid confusion or tech debt